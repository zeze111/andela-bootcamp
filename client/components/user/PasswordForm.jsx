import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { TextFieldGroup3 } from '../common/TextFieldGroup';

/**
 * @class PasswordForm
 * @extends {Component}
 */
class PasswordForm extends Component {
  /**
   * @description Constructor Function
   * @param {any} props
   * @memberof Home
   * @return {void}
   */
  constructor(props) {
    super(props);
    this.state = {
      oldPassword: '',
      newPassword: '',
    };
  }

  /**
   * @param {any} event
   * @memberof Home
   * @return {void}
   */
  onChange(event) {
    this.setState({ [event.target.name]: event.target.value });
  }

  /**
   * @memberof Home
   * @return {void}
   */
  render() {
    return (
      <div id="chpwd" className="col s12 m8 l7 push-l1 push-m1 form-style">
        <form className="col s10 push-s1">
          <TextFieldGroup3
            label="Old Password:"
            value={this.state.oldPassword}
            onChange={this.onChange}
            id="old"
            type="password"
            name="oldPassword"
            icon="lock_outline"
          />
          <TextFieldGroup3
            label="New Password:"
            value={this.state.newPassword}
            onChange={this.onChange}
            id="new"
            type="password"
            name="newPassword"
            icon="lock_outline"
          />
          <div className="right-align">
            <button className="btn grey" type="button"> Update </button>
          </div>
        </form>
      </div>
    );
  }
}

PasswordForm.propTypes = {
};

export default PasswordForm;
