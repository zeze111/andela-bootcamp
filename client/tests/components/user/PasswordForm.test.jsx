import React from 'react';

import PasswordForm from '../../../components/user/PasswordForm';
import store from '../../../store';

let props = {
  changePassword: jest.fn(() => Promise.resolve())
}

const setup = () => {
  const wrapper = shallow(<PasswordForm {...props} store={store} />);
  return { wrapper }
}

describe('onChange', () => {
  it('should change event', () => {
    const event = {
      target: { name: 'password', value: 'janedoe' },
    };
    const { wrapper } = setup();
    const password = wrapper.find('#old');
    password.simulate('change', event);
    expect(wrapper.instance().state.password).toBe('janedoe');
  });
});

describe('onSubmit', () => {
  it('be called when form is submitted', () => {
    const { wrapper } = setup();
    const event = {
      preventDefault: jest.fn(),
    };
    document.body.innerHTML =
    '<div>' +
    '  <form id="pwdForm" />' +
    '  <button id="button" />' +
    '</div>';
    wrapper.setState({
      oldPassword: 'janedoe',
      newPassword: 'janedo',
      newPassword_confirmation: 'janedo',
      errors: {},
      isLoading: false
    });

    wrapper.instance().onSubmit(event);
    expect(props.changePassword.mock.calls.length).toEqual(1);
  });
});

describe('onSubmit', () => {
  it('return validation error when form is submitted', () => {
    const { wrapper } = setup();
    const event = {
      preventDefault: jest.fn(),
    };
    document.body.innerHTML =
    '<div>' +
    '  <form id="pwdForm" />' +
    '  <button id="button" />' +
    '</div>';

    wrapper.setState({
      oldPassword: 'janedoe',
      newPassword: 'janedo',
    });
    wrapper.instance().onSubmit(event);
    expect(props.changePassword.mock.calls.length).toEqual(1);
  });
});

describe('onSubmit', () => {
  it('return an error when form is submitted', () => {
    const event = {
      preventDefault: jest.fn(),
    };
    props.changePassword = jest.fn(() => Promise.reject());
    const wrapper = shallow(<PasswordForm {...props} store={store} />);
    document.body.innerHTML =
    '<div>' +
    '  <form id="pwdForm" />' +
    '  <button id="button" />' +
    '</div>';

    wrapper.setState({
      oldPassword: 'janedoe',
      newPassword: 'janedo',
      newPassword_confirmation: 'janedo',
    });
    wrapper.instance().onSubmit(event);
    expect(props.changePassword.mock.calls.length).toEqual(1);
  });
});

describe('Test for Sign up form', () => {
  it('should render component', () => {

    const wrapper = setup();
    expect(wrapper).toMatchSnapshot()
  });
});
