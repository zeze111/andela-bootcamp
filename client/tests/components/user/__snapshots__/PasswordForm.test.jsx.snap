// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Sign up form should render component 1`] = `
Object {
  "wrapper": <div
    className="col s12 m8 l7 push-l1 push-m1 form-style"
    id="chpwd"
  >
    <form
      className="col s10 push-s1"
      id="pwdForm"
      onSubmit={[Function]}
    >
      <TextFieldGroup3
        error={Array []}
        icon="lock_outline"
        id="old"
        isDisabled={false}
        label="Old Password:"
        name="oldPassword"
        onChange={[Function]}
        type="password"
        value=""
      />
      <TextFieldGroup3
        error={Array []}
        icon="lock_outline"
        id="new"
        isDisabled={false}
        label="New Password:"
        name="newPassword"
        onChange={[Function]}
        type="password"
        value=""
      />
      <TextFieldGroup3
        error={Array []}
        icon="lock_outline"
        id="confirm"
        isDisabled={false}
        label="Confirm Password:"
        name="newPassword_confirmation"
        onChange={[Function]}
        type="password"
        value=""
      />
      <div
        className="right-align"
      >
        <input
          className="btn grey white-text"
          disabled={false}
          type="submit"
          value="Update"
        />
      </div>
      <span
        className="red-text error-text"
      />
    </form>
  </div>,
}
`;
